### ISSUE TITLE
`no-name-in-module` false positive on MacOS

### ISSUE THREAD

**Issue by OutOfFocus4** [PyCQA/pylint#4480]
I recently migrated a number of projects I work on from Python 3.7 to Python 3.9. After reinstalling `pylint` in the virtual environment, Visual Studio Code began showing errors saying that classes in the Python standard library didn't exist.

I have run into this issue with the following imports:

```py
from binascii import Error
from socket import error
from unicodedata import normalize
```

I encountered this issue on an iMac running MacOS 11.3.1 using CPython 3.9.5 installed via homebrew and pylint 2.8.2 and astroid 2.5.6. I tried to reproduce the issue on Windows, Linux (WSL 2), and FreeBSD, and failed.

### Steps to reproduce

<img width="1402" alt="pylint" src="https://user-images.githubusercontent.com/50265209/118494887-9994bc80-b6f0-11eb-93a3-04c8e17dafd9.png">

### Current behavior

`pylint` says the `binascii` module in the Python standard library has no `Error` member

### Expected behavior

`pylint` reports no errors

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.8.2
astroid 2.5.6
Python 3.9.5 (default, May  4 2021, 03:36:27)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

**Comment by cdce8p** [comment]
I just tried to reproduce it, but couldn't. I'm still using macOS Catalina though. I've used the installer from https://python.org.

For this file:
```py
from binascii import Error
from socket import error
from unicodedata import normalize
```

I get this output
```
************* Module test4
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: W0611: Unused Error imported from binascii (unused-import)
test.py:2:0: W0611: Unused error imported from socket (unused-import)
test.py:3:0: W0611: Unused normalize imported from unicodedata (unused-import)
```

```
pylint 2.8.2
astroid 2.5.6
Python 3.9.5 (v3.9.5:0a7dcbdb13, May  3 2021, 13:05:53) 
[Clang 12.0.5 (clang-1205.0.22.9)]
```

**Comment by OutOfFocus4** [comment]
> I just tried to reproduce it, but couldn't. I'm still using macOS Catalina though. I've used the installer from https://python.org.
> 
> For this file:
> 
> ```python
> from binascii import Error
> from socket import error
> from unicodedata import normalize
> ```
> 
> I get this output
> 
> ```
> ************* Module test
> test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
> test.py:1:0: W0611: Unused Error imported from binascii (unused-import)
> test.py:2:0: W0611: Unused error imported from socket (unused-import)
> test.py:3:0: W0611: Unused normalize imported from unicodedata (unused-import)
> ```
> 
> ```
> pylint 2.8.2
> astroid 2.5.6
> Python 3.9.5 (v3.9.5:0a7dcbdb13, May  3 2021, 13:05:53) 
> [Clang 12.0.5 (clang-1205.0.22.9)]
> ```

I installed the official release from https://python.org, and I couldn't reproduce it, either. It looks like the false positive is specific to the homebrew installation.

**Comment by cdce8p** [comment]
> I installed the official release from https://python.org, and I couldn't reproduce it, either. It looks like the false positive is specific to the homebrew installation.

As it is most likely related to Homebrew then, I'm going to close the issue.
